pallets/subtensor/tests/swap.rs

```diff
@@ -0,0 +1,1154 @@
+        );
+        assert_eq!(
+            SubtensorModule::get_delegate(new_hotkey.encode()),
+            SubtensorModule::get_delegate(old_hotkey.encode())
+        );
+        assert_eq!(
+            SubtensorModule::get_last_tx_block(&new_hotkey),
+            SubtensorModule::get_last_tx_block(&old_hotkey)
+        );
+
+        // Verify raw storage maps
+        // Stake
+        for (coldkey, stake_amount) in Stake::<Test>::iter_prefix(old_hotkey) {
+            assert_eq!(Stake::<Test>::get(new_hotkey, coldkey), stake_amount);
+        }
+
+        let mut weight = Weight::zero();
+        // UIDs
+        for netuid in SubtensorModule::get_netuid_is_member(&old_hotkey, &mut weight).unwrap() {
+            assert_eq!(
+                Uids::<Test>::get(netuid, new_hotkey),
+                Uids::<Test>::get(netuid, old_hotkey)
+            );
+        }
+
+        // Prometheus
+        for netuid in SubtensorModule::get_netuid_is_member(&old_hotkey, &mut weight).unwrap() {
+            assert_eq!(
+                Prometheus::<Test>::get(netuid, new_hotkey),
+                Prometheus::<Test>::get(netuid, old_hotkey)
+            );
+        }
+
+        // LoadedEmission
+        for netuid in SubtensorModule::get_netuid_is_member(&old_hotkey, &mut weight).unwrap() {
+            assert_eq!(
+                LoadedEmission::<Test>::get(netuid).unwrap(),
+                LoadedEmission::<Test>::get(netuid).unwrap()
+            );
+        }
+
+        // IsNetworkMember
+        for netuid in SubtensorModule::get_netuid_is_member(&old_hotkey, &mut weight).unwrap() {
+            assert!(IsNetworkMember::<Test>::contains_key(new_hotkey, netuid));
+            assert!(!IsNetworkMember::<Test>::contains_key(old_hotkey, netuid));
+        }
+
+        // Owner
+        assert_eq!(Owner::<Test>::get(new_hotkey), coldkey);
+
+        // TotalHotkeyStake
+        assert_eq!(
+            TotalHotkeyStake::<Test>::get(new_hotkey),
+            TotalHotkeyStake::<Test>::get(old_hotkey)
+        );
+
+        // Delegates
+        assert_eq!(
+            Delegates::<Test>::get(new_hotkey),
+            Delegates::<Test>::get(old_hotkey)
+        );
+
+        // LastTxBlock
+        assert_eq!(
+            LastTxBlock::<Test>::get(new_hotkey),
+            LastTxBlock::<Test>::get(old_hotkey)
+        );
+
+        // Axons
+        for netuid in SubtensorModule::get_netuid_is_member(&old_hotkey, &mut weight).unwrap() {
+            assert_eq!(
+                Axons::<Test>::get(netuid, new_hotkey),
+                Axons::<Test>::get(netuid, old_hotkey)
+            );
+        }
+
+        // TotalHotkeyColdkeyStakesThisInterval
+        assert_eq!(
+            TotalHotkeyColdkeyStakesThisInterval::<Test>::get(new_hotkey, coldkey),
+            TotalHotkeyColdkeyStakesThisInterval::<Test>::get(old_hotkey, coldkey)
+        );
+    });
+}
+
+#[test]
+fn test_do_swap_hotkey_ok_robust() {
```

Comment:
Do we need a validation for swapping Keys map too?

Reply:
done